### DISCUSSION TITLE
How to transform between RGB colorspaces?

### DISCUSSION THREAD

**Discussion by ahemberger** [colour-science/colour#1001] [Q&A]
Hi;

I have some data I've read using colour that I know to be in ACES-2065, and I'd like to move it to ACEScg. I thought I could do this thusly:

            img = colour.convert(img, "aces", "acescg")

but I see an error I'm having trouble digesting:

```
/Users/x/.pyenv/versions/3.9.13/lib/python3.9/site-packages/colour/utilities/verbose.py:262: ColourUsageWarning: The "Automatic Colour Conversion Graph" is a beta feature, be mindful of this when using it. Please report any unexpected behaviour and do not hesitate to ask any questions should they arise.
This warning can be disabled with the "colour.utilities.suppress_warnings" context manager as follows:
with colour.utilities.suppress_warnings(colour_usage_warnings=True): 
    convert(*args, **kwargs)
  warn(*args, **kwargs)
Traceback (most recent call last):
  File "/Users/x/projects/beautifulShelves/bin/bs_lgt_pano_tool.py", line 455, in <module>
    img = colour.convert(img, INSPACE, OUTSPACE)
  File "/Users/x/.pyenv/versions/3.9.13/lib/python3.9/site-packages/colour/graph/conversion.py", line 1255, in convert
    conversion_path = _conversion_path(source, target)
  File "/Users/x/.pyenv/versions/3.9.13/lib/python3.9/site-packages/colour/utilities/common.py", line 893, in wrapped
    return function(*args, **kwargs)
  File "/Users/x/.pyenv/versions/3.9.13/lib/python3.9/site-packages/colour/graph/conversion.py", line 900, in _conversion_path
    path = nx.shortest_path(CONVERSION_GRAPH, source, target)
  File "/Users/x/.pyenv/versions/3.9.13/lib/python3.9/site-packages/networkx/algorithms/shortest_paths/generic.py", line 165, in shortest_path
    paths = nx.bidirectional_shortest_path(G, source, target)
  File "/Users/x/.pyenv/versions/3.9.13/lib/python3.9/site-packages/networkx/algorithms/shortest_paths/unweighted.py", line 221, in bidirectional_shortest_path
    raise nx.NodeNotFound(msg)
networkx.exception.NodeNotFound: Either source aces or target acescg is not in G
```

**Comment by thomasmansencal** [comment]
Hi @ahemberger,

The colour conversion graph does not support that unfortunately, (it could though, maybe...). It is designed to convert between colour models, e.g. CIE XYZ to CIE LCHab. The conversion you need want is done with the [colour.RGB_to_RGB](https://colour.readthedocs.io/en/develop/generated/colour.RGB_to_RGB.html#colour.RGB_to_RGB) definition.

Cheers,

Thomas

**Reply by ahemberger** [reply]
Awesome, thank you sir.

A followup: The way I'm doing this at the moment is:

```
    (img, attribs) = colour.read_image(os.path.abspath(file), 
                            method="OpenImageIO", 
                            attributes=True )

    # do some stuff to img

    img = colour.RGB_to_RGB(img, 
                                colour.RGB_COLOURSPACES[INSPACE], 
                                colour.RGB_COLOURSPACES[OUTSPACE])

    colour.write_image(img, out_file,
                            method="OpenImageIO", 
                            attributes=attribs)
```

I can see that the colors in my image shift when I do this, though oiiotool reports that they both have the same chromaticities tag.

What's the most hygenic way to keep the headers in my EXR up to date with this transformation?

EDIT: Would it be:

`attribs['chromaticities'] = colour.RGB_COLOURSPACES[OUTSPACE].primaries`

?

**Reply by ahemberger** [reply]
Oh wait, I think I'm confused: is the exr chromaticities attribute meant to signify the chromaticities of the image data itself, or that of the colorspace in which the data is represented? 

I ask because I notice the chromaticities reported by oiiotool don't seem to exactly match those of:

```
aces = colour.models.RGB_COLOURSPACE_ACES2065_1
print(aces)
```

and now I'm wondering if I've misunderstood what that tag signifies.

**Reply by thomasmansencal** [reply]
The `chromaticities` attribute is meant to store both the primaries AND the whitepoint so you would need to concatenate them together first!
